Tensile/GenerateSummations.py

```diff
@@ -0,0 +1,201 @@
+def createLibraryForBenchmark(logicPath, libraryPath, currentPath):
+
+    pythonExePath = os.path.join(os.path.dirname(os.path.realpath(__file__)), "bin", "TensileCreateLibrary")
+    args = [pythonExePath, \
+        "--merge-files", "--no-legacy-components", \
+        "--new-client-only", "--no-short-file-names", "--no-library-print-debug", "--architecture=all", \
+        "--code-object-version=V3", "--cxx-compiler=hipcc", "--library-format=yaml", \
+        logicPath, libraryPath, "HIP"]
+
+    try:
+        subprocess.run(args, check=True, cwd=currentPath)
+    except (subprocess.CalledProcessError, OSError) as e:
+        printExit("ClientWriter Benchmark Process exited with error: {}".format(e))
+
+def GenerateSummations(userArgs):
+
+    inputLogicPath = userArgs[0]
+    outputPath = userArgs[1]
+    assignGlobalParameters({})
+
+    currentISA = globalParameters["CurrentISA"]
+    currentArchitecture = getArchitecture(currentISA)
+
+    globPath = os.path.join(inputLogicPath, "{}*".format(currentArchitecture))
+    logicFileNames = glob.glob(globPath)
+
+    for logicFileName in logicFileNames:
+
+        logicFileBaseName = os.path.basename(logicFileName)
+        logicFileStem, ext = os.path.splitext(logicFileBaseName)
+        if (ext != ".yaml"):
+            continue
+
+        currentPath = ensurePath(os.path.join(outputPath, logicFileStem))
+
+        libPath = ensurePath(os.path.join(currentPath, "lib"))
+        finalPath = ensurePath(os.path.join(currentPath, "final"))
+        localLogicPath = ensurePath(os.path.join(currentPath, "logic"))
+        localLogicFilePath = os.path.join(localLogicPath, logicFileBaseName)
+        
+        logic = LibraryIO.readLibraryLogicForSchedule(logicFileName)
+
+        (scheduleName, deviceNames, problemType, solutionsForSchedule, \
+           indexOrder, exactLogic, rangeLogic, _, architectureName) = logic
+
+        naming = Solution.getMinNaming(solutionsForSchedule)
+
+        for s in solutionsForSchedule:
+            s_state = s._state
+            name = Solution.getNameMin(s_state, naming)
+            s_state["SolutionNameMin"] = name          
+            isa = s_state["ISA"]
+            s_state["ISA"] = list(isa) 
+
+        exactLogic0 = {}
+        for e in exactLogic:
+            exactLogic0[tuple(e[0])] = e[1]
+
+        logicTuple = (problemType, solutionsForSchedule, indexOrder, exactLogic0, rangeLogic)
+        LibraryIO.configWriter("yaml").writeLibraryLogicForSchedule(localLogicPath, \
+            scheduleName, architectureName, \
+            deviceNames, logicTuple)
+
+        createLibraryForBenchmark(localLogicPath, libPath, currentPath)
+
+        logic1 = LibraryIO.readLibraryLogicForSchedule(localLogicFilePath)
+
+        (scheduleName1, deviceNames1, problemType1, solutionsForSchedule1, \
+           indexOrder1, exactLogic1, rangeLogic1, _, architectureName1) = logic1
+
+        exactList = []
```

Comment:
I am unclear why you would write a modified logic file and read it back in - can we avoid the IO by just using the modified logic tuple?

Reply:
This is a work around related to a bug in Tensile.

The first part of this is to generate consistent solution naming for both the bench marks generated by the client and the naming in the logic file.  Unfortunately in Tensile we have inconsistent typing for the DataType member of the ProblemType one version is a string and the other is an integer.  In the logic file this is set as a string and there is no problem reading it. However the process of generating the naming changes the type from string to integer. This is a problem for Tensile further down in the workflow which expects an the DataType field to be a string. To solve this would probably require some refactoring and a deeper investigation which is out of scope for this task.